otter/log/cloudfeeds.py

```diff
@@ -0,0 +1,173 @@
+    msg = event_copy["message"]
+    try:
+        event["message"] = msg[0]
+    except KeyError:
+        raise UnsuitableMessage("No message in event")
+
+    # map keys in event to CF keys
+    for log_key, cf_key in log_cf_mapping.iteritems():
+        if log_key in event:
+            cf_event[cf_key] = event[log_key]
+
+    if event.get('isError', False):
+        error = True
+        if ('traceback' in cf_event['message'] or
+           'exception' in cf_event['message']):
+            raise UnsuitableMessage(cf_event['message'])
+
+    return (cf_event, error,
+            datetime.utcfromtimestamp(event["time"]).isoformat())
+
+
+request_format = {
+    "entry": {
+        "@type": "http://www.w3.org/2005/Atom",
+        "title": "autoscale",
+        "content": {
+            "event": {
+                "@type": "http://docs.rackspace.com/core/event",
+                "id": "",
+                "version": "2",
+                "eventTime": "",
+                "type": "INFO",
+                "region": "",
+                "product": {
+                    "@type": "http://docs.rackspace.com/event/autoscale",
+                    "serviceCode": "Autoscale",
+                    "version": "1",
+                    "message": ""
+                }
+            }
+        }
+    }
+}
+
+
+def prepare_request(req_fmt, event, error, timestamp, region,
+                    _uuid=uuid.uuid4):
+    """
+    Prepare request based on request format
+    """
+    request = deepcopy(req_fmt)
+    if error:
+        request['entry']['content']['event']['type'] = 'ERROR'
+    request['entry']['content']['event']['region'] = region
+    request['entry']['content']['event']['eventTime'] = timestamp
+    request['entry']['content']['event']['product'].update(event)
+    request['entry']['content']['event']['id'] = str(_uuid())
+    return request
+
+
+def add_event(event, tenant_id, region, log,
+              _prep_req=prepare_request):
+    """
+    Add event to cloud feeds
+    # Review: Is taking _prep_req as argument really required?
+    """
+    event, error, timestamp = sanitize_event(event)
+    eff = retry_effect(
+        service_request(
+            ServiceType.CLOUD_FEEDS, 'POST',
+            append_segments('autoscale', 'events'),
+            data=_prep_req(request_format, event, error, timestamp, region),
+            log=log, success_pred=has_code(201)),
+        retry_times(5), exponential_backoff_interval(2))
+    return Effect(TenantScope(tenant_id=tenant_id, effect=eff))
+
+
+@attributes(['reactor', 'authenticator', 'tenant_id', 'region',
+             'service_configs', 'log', 'get_disp', 'add_event'],
+            defaults={'log': otter_log, 'get_disp': get_full_dispatcher,
+                      'add_event': add_event})
+class CloudFeedsObserver(object):
+    """
+    Log observer that pushes events to cloud feeds
+    """
+    def __init__(self):
+        # This is needed for attributes to work
```

Comment:
Yeah. I couldn't get this to work without the `__init__` :( It kept giving some error.